tests/test_highlighter.py

```diff
@@ -24,11 +65,23 @@ def test_wrong_type():
         ("repr.eui48", "ED-ED-ED-ED-ED-ED"),  # uppercase
         ("repr.eui48", "Ed-Ed-Ed-Ed-Ed-Ed"),  # mixed case
         ("repr.eui48", "0-00-1-01-2-02"),  # dropped zero
+        # Strings
+        ("repr.str", "'string'"),  # single quotes
+        ("repr.str", '"string"'),  # double quotes
+        ("repr.str", "'''string'''"),  # triple-single quotes
+        ("repr.str", '"""string"""'),  # triple-double quotes
+        # URLs
+        ("repr.url", "http://example.com/some/path"),  # http url
+        ("repr.url", "https://example.com/some/path"),  # https url
+        # UUIDs
+        ("repr.uuid", "00000000-0000-0000-0000-000000000000"),  # lowest uuid
+        ("repr.uuid", "12345678-90ab-cdef-1234-567890abcdef"),  # some uuid in between
+        ("repr.uuid", "ffffffff-ffff-ffff-ffff-ffffffffffff"),  # highest uuid
     ],
 )
 def test_highlight_regex(style_name: str, test_str: str):
     """Tests for the regular expressions used in ReprHighlighter."""
     text = Text(test_str)
     highlighter = ReprHighlighter()
     highlighter.highlight(text)
-    assert text._spans[-1] == Span(0, len(test_str), style_name)
+    assert style_name in repr(text)
```

Comment:
I should have probably added a comment to this affect, but originally I was testing the prevailing style (i.e. the last one) was the expected style, and that it covered the span of the entire string.

Otherwise if multiple regexes match then the wrong style could be applied on top. As an example, if the MAC address regexes are listed higher than the number regexes in the `highlights` list in `ReprHighlighter`, the individual numbers will be highlighted instead of the MAC address.

Reply:
I'm thinking the test should probably check that there is an appropriate span covering the entire string. Something along these lines:

```python
assert Span(0, len(test_str), style_name) in text.spans
```

Reply:
Perhaps we could add another test specifically for the more complex ones (like MAC address) to ensure it's highlighted as a MAC address rather than a sequence of numbers

Reply:
> I should have probably added a comment to this affect, but originally I was testing the prevailing style (i.e. the last one) was the expected style, and that it covered the span of the entire string.

I didn't think about it, but yes, that really is to be considered.
I think checking that there is a span covering the entire string won't work in every case, since there are matches that intentionally do not cover the whole string, like `<tag_start>`, `<tag_name>`, `<tag_contents>` and `<tag_end>`. So probably the best solution would really be to add a separate method for the more complex ones.

Reply:
@deppiedave64 good point.

Reply:
Maybe dividing it in one method for regexes that are supposed to match the whole string and one for those with multiple matches?

Reply:
@deppiedave64 Makes sense.

Reply:
All right, I will change that later. :+1:

Reply:
I've been looking in to the highlighter regexes in general, and I may have complicated matters. Hopefully for a good reason.

The regexes could overlap quite a bit, so something like an IP address would also be highlighted as two floating point numbers. There are a few other such issues such as the ipv6 regex also matching a valid EUI.

The output looks ok give the styles used, but it's inefficient as it creates pointless spans to process. In master I've rearranged the regexes, and grouped some of the regexes in to a single regex. Which should hopefully mean it doesn't create more spans than it needs to.

I may change it further after a bit of testing, so you might want to hold-off on this PR until I'm confident it's working correctly.

Thanks @deppiedave64 @Edward-Knight